src/functions_math.jl

```diff
@@ -31,6 +33,20 @@ for (NA, NB) in ((1,2), (2,1), (2,2))  #Vector * Vector, is not allowed
     end
 end
 
+# vector^T * vector
+function Base.:*(a::NamedDimsArray{A,T,2,<:CoVector}, b::NamedDimsArray{B,S,1}) where {A,B,T,S}
+    last(A) === first(B) || throw_matrix_dim_error(last(A), first(B))
+    return *(parent(a), parent(b))
+end
+
+function Base.:*(a::NamedDimsArray{L,T,2,<:CoVector}, b::AbstractVector) where {L,T}
+    return *(parent(a), b)
+end
+
+function Base.:*(a::CoVector, B::NamedDimsArray{L,T,1,<:AbstractVector}) where {L,T}
+    return *(a, parent(b))
+end
```

Comment:
If we remove this method, we still get an ambiguity, specifically:
```julia
  Expression: ndv' * ndv == adjoint(v) * ndv == transpose(v) * ndv
  MethodError: *(::Adjoint{Int64,Array{Int64,1}}, ::NamedDimsArray{(:vec,),Int64,1,Array{Int64,1}}) is ambiguous. Candidates:
    *(a::AbstractArray{T,2} where T, b::NamedDimsArray{#s19,#s20,1,A} where A<:AbstractArray{#s20,1} where #s20 where #s19) in NamedDims at /Users/npr/repos/NamedDims.jl/src/functions_math.jl:71
    *(u::Adjoint{T,#s623} where #s623<:(AbstractArray{T,1} where T) where T, v::AbstractArray{T,1} where T) in LinearAlgebra at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/adjtrans.jl:197
  Possible fix, define
    *(::Adjoint{T,#s623} where #s623<:(AbstractArray{T,1} where T) where T, ::NamedDimsArray{#s19,#s20,1,A} where A<:AbstractArray{#s20,1} where #s20 where #s19)
```

Reply:
I think this is now resolved... but in the most cumbersome way possible - just list the exactly all of the methods we want - and I don't love it

Reply:
`@eval` loop?

Reply:
seemed even less nice 🤷‍♂

Reply:
we use a lot of `@eval` loops

Reply:
okay.  we can write something like, if that seems nicer? :)
```julia
for A in (Adjoint{<:Any, <:AbstractVector}, Transpose{<:Real, <:AbstractVector{<:Real}})
    @eval function Base.:*(a::$A, b::NamedDimsArray{L,T,1,<:AbstractVector{T}}) where {L,T}
        return *(a, parent(b))
    end
end
```

Reply:
changed :)